src/main/java/com/acquia/http/HMACHttpRequestInterceptor.java

```diff
@@ -85,20 +85,29 @@ public void setCustomHeaders(String[] customHeaders) {
     }
 
     @Override
-    public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
+    public void process(HttpRequest request, HttpContext context)
+            throws HttpException, IOException {
         HMACAuthorizationHeader authHeader = this.createHMACAuthorizationHeader();
 
         HMACMessageCreator messageCreator = new HMACMessageCreator();
-        String message = messageCreator.createMessage(request, authHeader);
-        String encryptedMessage = "";
+        String signableRequestMessage = messageCreator.createSignableRequestMessage(request,
+            authHeader);
+        String signedRequestMessage = "";
         try {
-            encryptedMessage = this.algorithm.encryptMessage(this.secretKey, message);
+            signedRequestMessage = this.algorithm.encryptMessage(this.secretKey,
+                signableRequestMessage);
         } catch(SignatureException e) {
-            throw new IOException("Could not encrypt message", e);
+            throw new IOException("Fail to sign request message", e);
         }
 
-        authHeader.setSignature(encryptedMessage);
-        request.setHeader("Authorization", authHeader.toString()); //set it with encrypted signature
+        authHeader.setSignature(signedRequestMessage);
+        request.setHeader(HMACMessageCreator.PARAMETER_AUTHORIZATION, authHeader.toString()); //set it with encrypted signature
+
+        //set context for response interceptor
+        context.setAttribute("httpVerb", request.getRequestLine().getMethod().toUpperCase());
```

Comment:
These strings should be constants that are used in this class and the http response interceptor class